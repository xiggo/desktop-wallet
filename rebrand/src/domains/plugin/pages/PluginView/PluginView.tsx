import { LaunchRender } from "plugins";
import React, { useRef } from "react";
import { FallbackProps } from "react-error-boundary";

import { Page } from "@/app/components/Layout";
import { TextArea } from "@/app/components/TextArea";
import { useQueryParameters } from "@/app/hooks";
import { PluginImage } from "@/domains/plugin/components/PluginImage";
import { usePluginManagerContext } from "@/plugins/context/PluginManagerProvider";

const ErrorFallback = ({ error }: FallbackProps) => {
	const errorMessageReference = useRef();

	return (
		<div className="flex items-center py-10 w-full">
			<div className="container flex flex-col mx-auto">
				<h3 className="mb-4">An error occurred!</h3>

				<TextArea
					className="py-4"
					initialHeight={70}
					defaultValue={error.stack || error.message}
					ref={errorMessageReference}
					disabled
				/>
			</div>
		</div>
	);
};

export const PluginView = () => {
	const queryParameters = useQueryParameters();

	const { pluginManager } = usePluginManagerContext();

	const pluginId = queryParameters.get("pluginId")!;
	const plugin = pluginManager.plugins().findById(pluginId);

	/* istanbul ignore next */
	if (!plugin) {
		return <></>;
	}

	return (
		<Page>
			<div className="py-4 px-4 sm:px-6 lg:px-10">
				<div className="flex justify-between items-center">
					<div className="flex items-center space-x-3">
						<PluginImage size="3xs" logoURL={plugin.config().logo()} />

						<div className="flex space-x-10 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-700">
							<dl>
								<dd className="font-semibold text-theme-secondary-text dark:text-theme-text">
									{plugin.config().title()}
								</dd>
							</dl>
						</div>
					</div>
				</div>
			</div>

			<div className="flex relative flex-1 w-full h-full">
				<LaunchRender manager={pluginManager} pluginId={pluginId} fallback={ErrorFallback} />
			</div>
		</Page>
	);
};
