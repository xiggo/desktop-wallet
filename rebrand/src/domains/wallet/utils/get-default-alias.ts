import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
