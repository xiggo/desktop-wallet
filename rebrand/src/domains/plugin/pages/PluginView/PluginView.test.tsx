import { Contracts } from "@payvo/sdk-profiles";
import { LaunchPluginService } from "plugins";
import React from "react";
import { Route } from "react-router-dom";

import { PluginView } from "./PluginView";
import { PluginManagerProvider } from "@/plugins/context/PluginManagerProvider";
import { PluginController, PluginManager } from "@/plugins/core";
import { env, getDefaultProfileId, render, screen } from "@/utils/testing-library";

describe("Plugin View", () => {
	let manager: PluginManager;
	let profile: Contracts.IProfile;

	beforeEach(() => {
		profile = env.profiles().findById(getDefaultProfileId());
		manager = new PluginManager();
		manager.services().register([new LaunchPluginService()]);
	});

	afterAll(() => {
		jest.clearAllMocks();
	});

	it("should render plugin content", async () => {
		const plugin = new PluginController(
			{ "desktop-wallet": { permissions: ["LAUNCH"] }, name: "test-plugin" },
			(api) => api.launch().render(<h1>My Plugin View</h1>),
		);

		manager.plugins().push(plugin);

		plugin.enable(profile, { autoRun: true });

		const { container } = render(
			<Route path="/profiles/:profileId/plugins/view">
				<PluginManagerProvider manager={manager} services={[]}>
					<PluginView />
				</PluginManagerProvider>
			</Route>,
			{
				routes: [`/profiles/${profile.id()}/plugins/view?pluginId=${plugin.config().id()}`],
				withPluginProvider: false,
			},
		);

		await expect(screen.findByText("My Plugin View")).resolves.toBeVisible();

		expect(container).toMatchSnapshot();

		manager.plugins().removeById(plugin.config().id(), profile);
	});

	it.each(["message", "stack"])("should render error boundary with error %s", (type) => {
		const consoleSpy = jest.spyOn(console, "error").mockImplementation(() => {});

		const Component = () => {
			const error = new Error("error");

			if (type === "stack") {
				error.stack = "stacktrace";
			} else {
				error.stack = undefined;
			}

			throw error;
		};

		const plugin = new PluginController(
			{ "desktop-wallet": { permissions: ["LAUNCH"] }, name: "test-plugin" },
			(api) => api.launch().render(<Component />),
		);

		manager.plugins().push(plugin);

		plugin.enable(profile, { autoRun: true });

		const { container } = render(
			<Route path="/profiles/:profileId/plugins/view">
				<PluginManagerProvider manager={manager} services={[]}>
					<PluginView />
				</PluginManagerProvider>
			</Route>,
			{
				routes: [`/profiles/${profile.id()}/plugins/view?pluginId=${plugin.config().id()}`],
			},
		);

		expect(container).toHaveTextContent("An error occurred!");
		expect(container).toMatchSnapshot();

		manager.plugins().removeById(plugin.config().id(), profile);

		consoleSpy.mockRestore();
	});
});
