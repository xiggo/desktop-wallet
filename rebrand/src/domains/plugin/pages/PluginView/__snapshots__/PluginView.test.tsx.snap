// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin View should render error boundary with error message 1`] = `
<div>
  <div
    class="flex relative flex-col min-h-screen"
  >
    <nav
      aria-labelledby="main menu"
      class="NavigationBarblocks__NavWrapper-sc-nciu2a-0 iVVfMx"
    >
      <div
        class="flex relative h-21"
      >
        <button
          class="BackButton__StyledBackButton-sc-18ta3r7-0 gAKHms flex w-12"
        >
          <div
            class="sc-bdvvtL Cfvxc mx-auto"
            height="10"
            width="10"
          >
            <svg>
              chevron-left-small.svg
            </svg>
          </div>
        </button>
        <div
          class="flex flex-1 px-8"
        >
          <div
            class="flex items-center my-auto"
          >
            <div
              class="flex items-center justify-center my-auto mr-4 pl-0.5 text-white rounded-xl w-11 h-11 bg-theme-success-600"
            >
              <svg
                height="28"
              >
                payvo-logo.svg
              </svg>
            </div>
          </div>
          <ul
            class="flex mr-auto ml-4 space-x-8 h-21"
            data-testid="navigationBar__menu"
          >
            <li
              class="flex"
            >
              <a
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/dashboard"
                title="Portfolio"
              >
                Portfolio
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                aria-current="page"
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus active"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/plugins"
                title="Plugins"
              >
                Plugins
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/exchange"
                title="Exchange"
              >
                Exchange
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/news"
                title="News"
              >
                News
              </a>
            </li>
          </ul>
          <div
            class="flex items-center my-auto space-x-4"
            data-testid="navigationBar__notifications"
          >
            <div>
              <div
                class="relative"
              >
                <span
                  data-testid="dropdown__toggle"
                >
                  <div
                    class="NavigationBarblocks__NavigationButtonWrapper-sc-nciu2a-1 ijuJJT group"
                  >
                    <button
                      class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF"
                      data-testid="navbar__buttons--notifications"
                      type="button"
                    >
                      <div
                        class="flex items-center space-x-2"
                      >
                        <div
                          class="sc-bdvvtL qGXhc p-1"
                          height="20"
                          width="20"
                        >
                          <svg>
                            bell.svg
                          </svg>
                        </div>
                        <div
                          class="flex absolute top-1 right-1 justify-center items-center p-1 rounded-full transition-all duration-100 ease-linear bg-theme-background dark:group-hover:bg-theme-secondary-800 group-hover:bg-theme-primary-100"
                        >
                          <div
                            class="w-2 h-2 rounded-full bg-theme-danger-500"
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </span>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBarblocks__NavigationButtonWrapper-sc-nciu2a-1 ijuJJT"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF"
                  data-testid="navbar__buttons--send"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdvvtL qGXhc p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        double-arrow-right.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBarblocks__NavigationButtonWrapper-sc-nciu2a-1 ijuJJT"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF"
                  data-testid="navbar__buttons--receive"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdvvtL qGXhc p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        qr-code.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
          </div>
          <div
            class="flex items-center my-auto ml-8 space-x-4"
            data-testid="navigationBar__userInfo"
          >
            <div
              class="text-right"
            >
              <div
                class="text-xs font-semibold text-theme-secondary-500"
              >
                Your Balance
              </div>
              <div
                class="text-sm font-bold text-theme-secondary-text dark:text-theme-text"
                data-testid="Balance__value"
              >
                <span
                  class="whitespace-nowrap"
                  data-testid="Amount"
                >
                  0 BTC
                </span>
              </div>
            </div>
            <div
              class="relative"
            >
              <span
                data-testid="dropdown__toggle"
              >
                <div
                  class="relative justify-center items-center align-middle rounded-full cursor-pointer"
                  data-testid="navbar__useractions"
                >
                  <div
                    class="Avatar__AvatarWrapper-sc-18vw8a4-0 ewMeZp flex-shrink-0"
                    data-testid="Avatar"
                  >
                    <div
                      class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                    >
                      <img
                        alt="Profile Avatar"
                        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(233, 30, 99)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"80\\" cy=\\"30\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"55\\" cx=\\"0\\" cy=\\"60\\" fill=\\"rgb(255, 152, 0)\\"/><circle r=\\"40\\" cx=\\"50\\" cy=\\"50\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                      />
                      <span
                        class="absolute text-sm font-semibold text-theme-background dark:text-theme-text"
                      >
                        JO
                      </span>
                    </div>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="flex flex-col flex-1"
    >
      <div
        class="py-4 px-4 sm:px-6 lg:px-10"
      >
        <div
          class="flex justify-between items-center"
        >
          <div
            class="flex items-center space-x-3"
          >
            <div
              class="PluginImage__PluginImageWrapper-sc-10kztrf-0 cKUSGF bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700"
              data-testid="PluginImage"
            >
              <svg>
                plugin-logo-placeholder.svg
              </svg>
            </div>
            <div
              class="flex space-x-10 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-700"
            >
              <dl>
                <dd
                  class="font-semibold text-theme-secondary-text dark:text-theme-text"
                >
                  Test Plugin
                </dd>
              </dl>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex relative flex-1 w-full h-full"
      >
        <div
          class="flex items-center py-10 w-full"
        >
          <div
            class="container flex flex-col mx-auto"
          >
            <h3
              class="mb-4"
            >
              An error occurred!
            </h3>
            <div
              class="Input__InputWrapperStyled-sc-1i8sd1r-0 iJpvqJ py-4"
              disabled=""
            >
              <div
                class="relative flex flex-1 h-full"
              >
                <textarea
                  class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full resize-none text-theme-secondary-text"
                  data-testid="TextArea"
                  disabled=""
                  style="height: 4px;"
                >
                  error
                </textarea>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Plugin View should render error boundary with error stack 1`] = `
<div>
  <div
    class="flex relative flex-col min-h-screen"
  >
    <nav
      aria-labelledby="main menu"
      class="NavigationBarblocks__NavWrapper-sc-nciu2a-0 iVVfMx"
    >
      <div
        class="flex relative h-21"
      >
        <button
          class="BackButton__StyledBackButton-sc-18ta3r7-0 gAKHms flex w-12"
        >
          <div
            class="sc-bdvvtL Cfvxc mx-auto"
            height="10"
            width="10"
          >
            <svg>
              chevron-left-small.svg
            </svg>
          </div>
        </button>
        <div
          class="flex flex-1 px-8"
        >
          <div
            class="flex items-center my-auto"
          >
            <div
              class="flex items-center justify-center my-auto mr-4 pl-0.5 text-white rounded-xl w-11 h-11 bg-theme-success-600"
            >
              <svg
                height="28"
              >
                payvo-logo.svg
              </svg>
            </div>
          </div>
          <ul
            class="flex mr-auto ml-4 space-x-8 h-21"
            data-testid="navigationBar__menu"
          >
            <li
              class="flex"
            >
              <a
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/dashboard"
                title="Portfolio"
              >
                Portfolio
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                aria-current="page"
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus active"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/plugins"
                title="Plugins"
              >
                Plugins
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/exchange"
                title="Exchange"
              >
                Exchange
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/news"
                title="News"
              >
                News
              </a>
            </li>
          </ul>
          <div
            class="flex items-center my-auto space-x-4"
            data-testid="navigationBar__notifications"
          >
            <div>
              <div
                class="relative"
              >
                <span
                  data-testid="dropdown__toggle"
                >
                  <div
                    class="NavigationBarblocks__NavigationButtonWrapper-sc-nciu2a-1 ijuJJT group"
                  >
                    <button
                      class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF"
                      data-testid="navbar__buttons--notifications"
                      type="button"
                    >
                      <div
                        class="flex items-center space-x-2"
                      >
                        <div
                          class="sc-bdvvtL qGXhc p-1"
                          height="20"
                          width="20"
                        >
                          <svg>
                            bell.svg
                          </svg>
                        </div>
                        <div
                          class="flex absolute top-1 right-1 justify-center items-center p-1 rounded-full transition-all duration-100 ease-linear bg-theme-background dark:group-hover:bg-theme-secondary-800 group-hover:bg-theme-primary-100"
                        >
                          <div
                            class="w-2 h-2 rounded-full bg-theme-danger-500"
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </span>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBarblocks__NavigationButtonWrapper-sc-nciu2a-1 ijuJJT"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF"
                  data-testid="navbar__buttons--send"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdvvtL qGXhc p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        double-arrow-right.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBarblocks__NavigationButtonWrapper-sc-nciu2a-1 ijuJJT"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF"
                  data-testid="navbar__buttons--receive"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdvvtL qGXhc p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        qr-code.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
          </div>
          <div
            class="flex items-center my-auto ml-8 space-x-4"
            data-testid="navigationBar__userInfo"
          >
            <div
              class="text-right"
            >
              <div
                class="text-xs font-semibold text-theme-secondary-500"
              >
                Your Balance
              </div>
              <div
                class="text-sm font-bold text-theme-secondary-text dark:text-theme-text"
                data-testid="Balance__value"
              >
                <span
                  class="whitespace-nowrap"
                  data-testid="Amount"
                >
                  0 BTC
                </span>
              </div>
            </div>
            <div
              class="relative"
            >
              <span
                data-testid="dropdown__toggle"
              >
                <div
                  class="relative justify-center items-center align-middle rounded-full cursor-pointer"
                  data-testid="navbar__useractions"
                >
                  <div
                    class="Avatar__AvatarWrapper-sc-18vw8a4-0 ewMeZp flex-shrink-0"
                    data-testid="Avatar"
                  >
                    <div
                      class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                    >
                      <img
                        alt="Profile Avatar"
                        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(233, 30, 99)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"80\\" cy=\\"30\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"55\\" cx=\\"0\\" cy=\\"60\\" fill=\\"rgb(255, 152, 0)\\"/><circle r=\\"40\\" cx=\\"50\\" cy=\\"50\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                      />
                      <span
                        class="absolute text-sm font-semibold text-theme-background dark:text-theme-text"
                      >
                        JO
                      </span>
                    </div>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="flex flex-col flex-1"
    >
      <div
        class="py-4 px-4 sm:px-6 lg:px-10"
      >
        <div
          class="flex justify-between items-center"
        >
          <div
            class="flex items-center space-x-3"
          >
            <div
              class="PluginImage__PluginImageWrapper-sc-10kztrf-0 cKUSGF bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700"
              data-testid="PluginImage"
            >
              <svg>
                plugin-logo-placeholder.svg
              </svg>
            </div>
            <div
              class="flex space-x-10 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-700"
            >
              <dl>
                <dd
                  class="font-semibold text-theme-secondary-text dark:text-theme-text"
                >
                  Test Plugin
                </dd>
              </dl>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex relative flex-1 w-full h-full"
      >
        <div
          class="flex items-center py-10 w-full"
        >
          <div
            class="container flex flex-col mx-auto"
          >
            <h3
              class="mb-4"
            >
              An error occurred!
            </h3>
            <div
              class="Input__InputWrapperStyled-sc-1i8sd1r-0 iJpvqJ py-4"
              disabled=""
            >
              <div
                class="relative flex flex-1 h-full"
              >
                <textarea
                  class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full resize-none text-theme-secondary-text"
                  data-testid="TextArea"
                  disabled=""
                  style="height: 4px;"
                >
                  stacktrace
                </textarea>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Plugin View should render plugin content 1`] = `
<div>
  <div
    class="flex relative flex-col min-h-screen"
  >
    <nav
      aria-labelledby="main menu"
      class="NavigationBarblocks__NavWrapper-sc-nciu2a-0 iVVfMx"
    >
      <div
        class="flex relative h-21"
      >
        <button
          class="BackButton__StyledBackButton-sc-18ta3r7-0 gAKHms flex w-12"
        >
          <div
            class="sc-bdvvtL Cfvxc mx-auto"
            height="10"
            width="10"
          >
            <svg>
              chevron-left-small.svg
            </svg>
          </div>
        </button>
        <div
          class="flex flex-1 px-8"
        >
          <div
            class="flex items-center my-auto"
          >
            <div
              class="flex items-center justify-center my-auto mr-4 pl-0.5 text-white rounded-xl w-11 h-11 bg-theme-success-600"
            >
              <svg
                height="28"
              >
                payvo-logo.svg
              </svg>
            </div>
          </div>
          <ul
            class="flex mr-auto ml-4 space-x-8 h-21"
            data-testid="navigationBar__menu"
          >
            <li
              class="flex"
            >
              <a
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/dashboard"
                title="Portfolio"
              >
                Portfolio
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                aria-current="page"
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus active"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/plugins"
                title="Plugins"
              >
                Plugins
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/exchange"
                title="Exchange"
              >
                Exchange
              </a>
            </li>
            <li
              class="flex"
            >
              <a
                class="flex relative items-center font-semibold transition-colors duration-200 focus:outline-none text-md text-theme-secondary-text ring-focus"
                data-ring-focus-margin="-mx-2"
                href="/profiles/b999d134-7a24-481e-a95d-bc47c543bfc9/news"
                title="News"
              >
                News
              </a>
            </li>
          </ul>
          <div
            class="flex items-center my-auto space-x-4"
            data-testid="navigationBar__notifications"
          >
            <div>
              <div
                class="relative"
              >
                <span
                  data-testid="dropdown__toggle"
                >
                  <div
                    class="NavigationBarblocks__NavigationButtonWrapper-sc-nciu2a-1 ijuJJT group"
                  >
                    <button
                      class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF"
                      data-testid="navbar__buttons--notifications"
                      type="button"
                    >
                      <div
                        class="flex items-center space-x-2"
                      >
                        <div
                          class="sc-bdvvtL qGXhc p-1"
                          height="20"
                          width="20"
                        >
                          <svg>
                            bell.svg
                          </svg>
                        </div>
                        <div
                          class="flex absolute top-1 right-1 justify-center items-center p-1 rounded-full transition-all duration-100 ease-linear bg-theme-background dark:group-hover:bg-theme-secondary-800 group-hover:bg-theme-primary-100"
                        >
                          <div
                            class="w-2 h-2 rounded-full bg-theme-danger-500"
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </span>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBarblocks__NavigationButtonWrapper-sc-nciu2a-1 ijuJJT"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF"
                  data-testid="navbar__buttons--send"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdvvtL qGXhc p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        double-arrow-right.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBarblocks__NavigationButtonWrapper-sc-nciu2a-1 ijuJJT"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF"
                  data-testid="navbar__buttons--receive"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdvvtL qGXhc p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        qr-code.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
          </div>
          <div
            class="flex items-center my-auto ml-8 space-x-4"
            data-testid="navigationBar__userInfo"
          >
            <div
              class="text-right"
            >
              <div
                class="text-xs font-semibold text-theme-secondary-500"
              >
                Your Balance
              </div>
              <div
                class="text-sm font-bold text-theme-secondary-text dark:text-theme-text"
                data-testid="Balance__value"
              >
                <span
                  class="whitespace-nowrap"
                  data-testid="Amount"
                >
                  0 BTC
                </span>
              </div>
            </div>
            <div
              class="relative"
            >
              <span
                data-testid="dropdown__toggle"
              >
                <div
                  class="relative justify-center items-center align-middle rounded-full cursor-pointer"
                  data-testid="navbar__useractions"
                >
                  <div
                    class="Avatar__AvatarWrapper-sc-18vw8a4-0 ewMeZp flex-shrink-0"
                    data-testid="Avatar"
                  >
                    <div
                      class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                    >
                      <img
                        alt="Profile Avatar"
                        src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(233, 30, 99)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"80\\" cy=\\"30\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"55\\" cx=\\"0\\" cy=\\"60\\" fill=\\"rgb(255, 152, 0)\\"/><circle r=\\"40\\" cx=\\"50\\" cy=\\"50\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                      />
                      <span
                        class="absolute text-sm font-semibold text-theme-background dark:text-theme-text"
                      >
                        JO
                      </span>
                    </div>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="flex flex-col flex-1"
    >
      <div
        class="py-4 px-4 sm:px-6 lg:px-10"
      >
        <div
          class="flex justify-between items-center"
        >
          <div
            class="flex items-center space-x-3"
          >
            <div
              class="PluginImage__PluginImageWrapper-sc-10kztrf-0 cKUSGF bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700"
              data-testid="PluginImage"
            >
              <svg>
                plugin-logo-placeholder.svg
              </svg>
            </div>
            <div
              class="flex space-x-10 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-700"
            >
              <dl>
                <dd
                  class="font-semibold text-theme-secondary-text dark:text-theme-text"
                >
                  Test Plugin
                </dd>
              </dl>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex relative flex-1 w-full h-full"
      >
        <h1>
          My Plugin View
        </h1>
      </div>
    </div>
  </div>
</div>
`;
